uat/custom-components/client-python-paho/src/mqtt_lib/temp_files_manager.py

```diff
@@ -0,0 +1,36 @@
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+"""Manager for working with temp files"""
+
+import tempfile
+import os
+
+
+class TempFilesManager:
+    """Manager for working with temp files"""
+
+    def __init__(self):
+        """Construct TempFilesManager"""
+        self.__temp_files = []
+
+    def create_new_temp_file(self, data: str) -> str:
+        """
+        Create new temp file
+        Parameters:
+        ----------
+        Returns new temp file path in system
+        """
+        new_file = tempfile.NamedTemporaryFile(mode="wt", delete=False)  # pylint: disable=consider-using-with

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *new\_file*. The resource is allocated by call *tempfile.NamedTemporaryFile*. The resource is referenced at lines 27, 28, 29, 31. The resource is closed at line 28. There are other execution paths that do not contain closure statements, for example, when *\_TemporaryFileWrapper.write()* throws an exception. To prevent this resource leak, close *new\_file* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).

Reply:
fixed in e511535